quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -100,7 +103,34 @@ public void beforeEach(ExtensionContext context) throws Exception {
             onBeforeStartDistribution(context.getRequiredTestMethod().getAnnotation(BeforeStartDistribution.class));
 
             if (launch != null) {
-                result = dist.run(Arrays.asList(launch.value()));
+                List<String> argsFromLaunchAnnotation = new ArrayList<>(Arrays.asList(launch.value()));
+
+                WithLegacyStoreOnly withLegacyStoreOnly = context.getRequiredTestMethod().getAnnotation(WithLegacyStoreOnly.class);
+                if (withLegacyStoreOnly == null) {
+                    withLegacyStoreOnly = context.getTestClass()
+                            .orElse(Object.class)
+                            .getDeclaredAnnotation(WithLegacyStoreOnly.class);
+                }
+
+                Boolean useCHMStorageWhenPossible = Boolean.parseBoolean(System.getProperty("kc.test.storage.use-chm-storage-when-possible", "true"));
```

Comment:
What does it mean exactly to use `chm` when possible? Do we really need this flag?